### ISSUE TITLE
Not an issiue, but a suggestion

### ISSUE THREAD

**Issue by Winkenbm** [SoftXperience/home-assistant-foxess-api#1]
Hello,
first of all, thank you very much for your efforts. Your integration works great. The only thing I'm missing is that the sensors can be used in the energy dashboard. Perhaps that would be a suggestion for the future. 
Good luck with your development.
Kind regards,
Michael

**Comment by Winkenbm** [comment]
Sorry, i think i found my mistake.

**Comment by SoftXperience** [comment]
Yes, sensors should have correct state and device classes and work well in my energy dashboard.

**Comment by Winkenbm** [comment]
The sensors work fine, much better than the old integration i used.
For my understanding, the Energy Dashboard needs the current solar production and the current grid consumption/feed in, battery charge/discharge. But when i try to add the sensors to the dashboard, i only  see the daily or total sensors:

Sensors showing in the Energy Dashboard:
- Daily grid consumption energy
- Daily feedin energy
- Daily generated energy
- Charged Energy Total
- Discharged Energy Total

What i think i need:
- Grid Consumption Power
- Feed-in Power
- PV Power
- Battery Charge Power
- Battery Discharge Power

Right now my dashboard shows no energy flows. Do you have any suggestions for me?

**Comment by SoftXperience** [comment]
For solar production use the _generation (cumulative generation) sensor. It returns the overall sum of your production and the energy dashboard takes care of the calculations.
For battery use _charged_energy_total and _discharged_energy_total. These are on a daily basis, as I can't find an overall cumulative value for that in the API. I just renamed them to _daily_... to better reflect their content. Please update to 1.0.7.
As their name now implies, they reset every day. But as the state_class is correctly used, the energy dashboard should be fine with that.

For grid consumption and feedin I use the sensors of my smart meter reader directly. But you should be fine using daily_grid_consumption and daily_feedin.

The power sensors you mentioned are realtime measurements and are not suitable for the dashboard.

I will add that to the readme.md

**Comment by Winkenbm** [comment]
This is my config right now:
![image](https://github.com/SoftXperience/home-assistant-foxess-api/assets/50736123/683aa220-f44a-4a4e-95fb-90a9f4f5f832)

And this is what my Dashboard shows:
![image](https://github.com/SoftXperience/home-assistant-foxess-api/assets/50736123/b4b29a7c-f1e7-49c7-a3c7-a7ced79cdfc8)

I hope the dashboard will fill with time.

**Comment by SoftXperience** [comment]
I updated the readme. Use sensor.<serial>_generation for solar production. The daily value seems to hang behind a bit... don't know why.
The other config looks good to me and should fill the dashboard over time...

**Comment by SoftXperience** [comment]
![Bildschirmfoto 2024-02-01 um 11 51 36](https://github.com/SoftXperience/home-assistant-foxess-api/assets/8452407/a98fc9f7-8d05-4687-9754-951be8a5c972)
That's what I see

**Comment by SoftXperience** [comment]
I'm not sure if the charged battery value is correct or if I need to switch it to positive value.

**Comment by Winkenbm** [comment]
I will have a look. Btw. german dashboard. Than we can continue in german. Macht es leichter ;-)

**Comment by SoftXperience** [comment]
Yes, german would be ok, too. But if there will be other people having similar problems, they can read the existing issues.

I had a look, the value of the charged energy is positive, but the dashboard shows it as negative. Don't quite understand that yet, but I guess it would be fine...

**Comment by SoftXperience** [comment]
Nochmal eine Frage aus Interesse: wie hast du meine Integration gefunden? Ich würde sie gern direkt in HA bringen, aber die Hürden dafür sind recht hoch (an Code, Doku usw). Und weil ich es natürlich selbst nutze, wollte ich es erstmal schnell lauffähig bekommen...
Aber ich glaube, es gibt auch keine Möglichkeit herauszufinden, wieviele Leute es nutzen. Wäre natürlich interessant...

**Comment by Winkenbm** [comment]
Sorry für die späte Antwort. Ich habe ganz stumpf auf GitHub nach FoxESS gesucht. Dein Ansatz schien mir logisch, besser die API als die Webseite zu nutzen. Vor allem da die Webseite oft nicht erreichbar war. 
Eines ist mir heute aufgefallen wo ich noch nicht genau weiß woran es liegt. Meine Solar-Produktion startete heute stark nach hinten versetzt laut Energy Dashboard, ging dafür aber bis nach 21 Uhr. Da hat die Sonne mit Sicherheit nicht mehr bei uns geliefert.... 
Mal schauen wie es morgen ist.

**Comment by neustaTerasa** [comment]
Ok, wäre halt spannend zu wissen, wieviele das nutzen. Aber solange ich es auch selbst nutze, werde ich es wohl sowieso weiter pflegen.

Bzgl der Produktion, hast du mal die Werte mit denen aus der App oder der Webseite verglichen?
Ich habe da gerade bei den kumulativen Werten (tägliche Verbräuche) auch leicht abweichende Daten festgestellt. Ich weiß aber natürlich auch nicht genau, welche Daten die App/Webseite nutzen und ob die noch andere Verarbeitungen durchführen. Ich reiche hier ja nur die Daten der API durch.

Bzw hast du auch mal geschaut, wie oft dein Datalogger überträgt? Sensor last_update ist das Datum, was die Schnittstelle überträgt. Ich gehe mal davon aus, dass es der Zeitstempel ist, wann die Cloud Daten empfangen hat.

**Comment by Winkenbm** [comment]
So, es scheint sich alles eingependelt zu haben, die Daten sind recht genau. Was mich zuerst gewundert hat, war, dass Verbrauch aus der Batterie auch als Solarproduktion angezeigt wird. Im nachhinein macht es aber sinn, da die Batterie ja mit Solarstrom geladen wurde. Wenn mann das jetzt noch farblich trennen könnte wäre es super, aber nötig ist es nicht.
Danke für deine Arbeit!

**Comment by SoftXperience** [comment]
Meinst du das in der kleinen Grafik mit den fließenden Punkten? Weil in dem Balkendiagramm vom Tag sind bei mir die Anteile vom Verbrauch durchaus in den entsprechenden Farben.
In dem kleinen Bild geht es da glaube ich um die Autarkie und da hast du es ja schon gesagt, die Energie aus der Batterie wurde ja vorher auch selbst erzeugt.

Ansonsten liegen diese Sachen natürlich nicht mehr im Bereich meiner Integration, sondern in Home Assistant selbst.

**Comment by SoftXperience** [comment]
Also bei mir werden die Verbräuche korrekt aufgeteilt und auch der Strom aus dem Speicher wird extra ausgewiesen und angezeigt. Sowohl im Balkendiagramm, als auch in dem kleinen...
![Screenshot_20240210-150254~2](https://github.com/SoftXperience/home-assistant-foxess-api/assets/8452407/be88be11-1d8c-4b96-bfb3-5e1a8235500a)
![Screenshot_20240210-150257~2](https://github.com/SoftXperience/home-assistant-foxess-api/assets/8452407/854520d6-f9bc-47dd-a3b5-65de00d9ad8e)
